relbench/external/nn.py

```diff
@@ -167,17 +167,19 @@ def forward(
         num_sampled_edges_dict: Optional[Dict[EdgeType, List[int]]] = None,
     ) -> Dict[NodeType, Tensor]:
         for i, (conv, norm_dict) in enumerate(zip(self.convs, self.norms)):
+            # Removing trimming since it does not pass
```

Comment:
What's the problem with trimming?

Reply:
`x_dict` becomes empty when edge_index_dict is empty.

Reply:
```
    def test_node_train_empty_graph(tmp_path):
        # Make a very sparse graph
        num_customers = 50
        dataset = FakeDataset(num_customers=num_customers, num_reviews=1)
    
        data, col_stats_dict = make_pkey_fkey_graph(
            dataset.db,
            get_stype_proposal(dataset.db),
            text_embedder_cfg=TextEmbedderConfig(
                text_embedder=HashTextEmbedder(8), batch_size=None
            ),
            cache_dir=tmp_path,
        )
        node_to_col_names_dict = {  # TODO Expose as method in `HeteroData`.
            node_type: data[node_type].tf.col_names_dict for node_type in data.node_types
        }
        loader = NeighborLoader(
            data,
            num_neighbors=[-1, -1],
            time_attr="time",
            input_nodes=("customer", torch.arange(num_customers)),
            input_time=torch.zeros(num_customers, dtype=torch.long),
            batch_size=5,
            # Only shuffle during training.
            shuffle=True,
        )
        encoder = HeteroEncoder(64, node_to_col_names_dict, col_stats_dict)
        gnn = HeteroGraphSAGE(data.node_types, data.edge_types, 64)
        head = MLP(64, out_channels=1, num_layers=1)
    
        for batch in loader:
            x_dict = encoder(batch.tf_dict)
            x_dict = gnn(
                x_dict,
                batch.edge_index_dict,
                batch.num_sampled_nodes_dict,
                batch.num_sampled_edges_dict,
            )
>           head(x_dict["customer"])
E           KeyError: 'customer'

test_node_nn.py:185: KeyError
========================================= short test summary info ==========================================
FAILED test_node_nn.py::test_node_train_empty_graph - KeyError: 'customer'
```

Reply:
Still wondering though why seed node type should be empty based on this. It should always contain the seed nodes at least.

Reply:
Thanks, feel free to delete the line once fixed on PyG side.

Can you try running the test and help debug?

Reply:
Can I merge this to unblock the team? It's needed to run Amazon recommendation.

Reply:
Issue tracked [here](https://github.com/pyg-team/pytorch_geometric/issues/9015).